### PR TITLE
NT-1918: Create GraphQL Query & Fragments

### PR DESCRIPTION
**Pull request by sunday-okpoluaefe**
# 📲 What

Create GraphQL Query & Fragments

# 🤔 Why

- Loads the first 25 Comments

- Graph QL Query and Fragments are created

# 🛠 How

- Created graphql fragmets
```
fragment comment on Comment {
    id
    author {
       ...user
    }
    body
    deleted
    parentId
    createdAt
}

```

```
fragment user on User {
    name
    id
    imageUrl(blur: false, width: 54),
    isCreator
}
```

```
fragment pageInfo on PageInfo {
      hasPreviousPage
      hasNextPage
      startCursor
      endCursor
}

```

- Created comments query
```
query GetProjectComments($slug: String!, $cursor: String!){
      project(slug: $slug) {
        collaborators {
          edges {
            node {
              id
              name
            }
          }
        }
        comments(after: $cursor) {
          edges {
            cursor
            node {
              authorBadges
              ...comment
              replies {
                edges {
                cursor
                  node {
                    ...comment
                  }
                }
                pageInfo {
                    ...pageInfo
                 }
                totalCount
              }
            }
          }
          pageInfo {
                  ...pageInfo
           }
          totalCount
        }
      }
   }

```

# 👀 See

Trello, screenshots, external resources?

| Before 🐛 | After 🦋 |
| --- | --- |
|  |  |

# 📋 QA

QA -> You can execute this query directly on https://staging.kickstarter.com/graphiql

```
query {
	project(slug: "nimble-salon-quality-nails-from-the-comfort-of-your-home") {
    collaborators {
            edges {
              node {
                id
                name
              }
            }
          }
      comments {
        edges {
          cursor
          node {
            authorBadges
            ...comment
            replies {
              edges {
              cursor
                node {
                  ...comment
                }
              }
              pageInfo {
                  ...pageInfo
               }
              totalCount
            }
          }
        }
        pageInfo {
                ...pageInfo
         }
        totalCount
      }
    }
 }

fragment user on User {
    name
    id
    imageUrl(blur: false, width: 54),
    isCreator
}

fragment pageInfo on PageInfo {
      hasPreviousPage
      hasNextPage
      startCursor
      endCursor
}

fragment comment on Comment {
    id
    author {
       ...user
    }
    body
    deleted
    parentId
    createdAt
}
```

# Story 📖

https://kickstarter.atlassian.net/browse/NT-1918

### CODE REVIEW

File: app/src/main/graphql/project.graphql [hunk 1]
```diff
@@ -32,3 +32,42 @@ query GetProjectAddOns($slug: String!, $locationId: ID!) {
...
+              name
+            }
+          }
+        }
+        comments(after: $cursor) {
+          edges {
+            cursor
+            node {
+              authorBadges
+              ...comment
```
Comment (Arkariang) [inline app/src/main/graphql/project.graphql:51]: Great work with that fragment 👍

Comment (Arkariang) [reply app/src/main/graphql/project.graphql:51]: small question here.
What do you think about moving `authorBadges` inside the comment fragment?. If I'm correct the badges are those tags like : "Superbaker" in the designs those tags will be diplayed in the comment card, so I guess it makes sense considering the badges as comment field

**Review by Arkariang — COMMENTED** [review]
As QA I executed the query on https://staging.kickstarter.com/graphiql and the result of it is the list of comments and the collaborators. Not sure though why we need to collaborators, but if that is also in the acceptance criteria for this ticket disregard my comment and I'll approve it :).

QA -> You can execute this query directly on https://staging.kickstarter.com/graphiql 

```
query {
	project(slug: "nimble-salon-quality-nails-from-the-comfort-of-your-home") {
    collaborators {
            edges {
              node {
                id
                name
              }
            }
          }
      comments {
        edges {
          cursor
          node {
            authorBadges
            ...comment
            replies {
              edges {
              cursor
                node {
                  ...comment
                }
              }
              pageInfo {
                  ...pageInfo
               }
              totalCount
            }
          }
        }
        pageInfo {
                ...pageInfo
         }
        totalCount
      }
    }
 }

fragment user on User {
    name
    id
    imageUrl(blur: false, width: 54),
    isCreator
}

fragment pageInfo on PageInfo {
      hasPreviousPage
      hasNextPage
      startCursor
      endCursor
}

fragment comment on Comment {
    id
    author {
       ...user
    }
    body
    deleted
    parentId
    createdAt
}
```

**Comment by sunday-okpoluaefe** [comment]
> As QA I executed the query on https://staging.kickstarter.com/graphiql and the result of it is the list of comments and the collaborators. Not sure though why we need the collaborators, but if that is also in the acceptance criteria for this ticket disregard my comment and I'll approve it :).
> You've done a great work here @sunday-okpoluaefe
> 
> QA -> You can execute this query directly on https://staging.kickstarter.com/graphiql
> 
> ```
> query {
> 	project(slug: "nimble-salon-quality-nails-from-the-comfort-of-your-home") {
>     collaborators {
>             edges {
>               node {
>                 id
>                 name
>               }
>             }
>           }
>       comments {
>         edges {
>           cursor
>           node {
>             authorBadges
>             ...comment
>             replies {
>               edges {
>               cursor
>                 node {
>                   ...comment
>                 }
>               }
>               pageInfo {
>                   ...pageInfo
>                }
>               totalCount
>             }
>           }
>         }
>         pageInfo {
>                 ...pageInfo
>          }
>         totalCount
>       }
>     }
>  }
> 
> fragment user on User {
>     name
>     id
>     imageUrl(blur: false, width: 54),
>     isCreator
> }
> 
> fragment pageInfo on PageInfo {
>       hasPreviousPage
>       hasNextPage
>       startCursor
>       endCursor
> }
> 
> fragment comment on Comment {
>     id
>     author {
>        ...user
>     }
>     body
>     deleted
>     parentId
>     createdAt
> }
> ```

thanks. For the collaborators, I just feel there will be need to have list of participants in the comment. However, I can remove them for now.

**Review by Arkariang — APPROVED** [review]

**Pull request merged**